cpp/arcticdb/processing/clause.cpp

```diff
@@ -784,7 +784,11 @@ std::shared_ptr<Column> ResampleClause<closed_boundary>::generate_output_index_c
     for (const auto& input_index_column: input_index_columns) {
         auto index_column_data = input_index_column->data();
         const auto cend = index_column_data.cend<IndexTDT>();
-        for (auto it = index_column_data.cbegin<IndexTDT>(); it != cend; ++it) {
+        auto it = index_column_data.cbegin<IndexTDT>();
+        while (it != cend && *it < date_range_->first) {
+            ++it;
+        }
+        for (;it != cend && *it <= date_range_->second; ++it) {
```

Comment:
Add some comments explaining which corner cases are dealt with by which bits of code here